File: sdks/typescript/examples/basic-chat.ts
/**
 * basic-chat.ts — Synchronous chat example.
 *
 * Sends a single message and waits for the complete response.
 *
 * Run with:
 *   npx tsx examples/basic-chat.ts
 */

import { SuperagentClient } from "../src/index.js";

const client = new SuperagentClient({
  baseUrl: process.env["SUPERAGENT_URL"] ?? "http://localhost:8888",
  // apiKey: process.env['SUPERAGENT_API_KEY'],
});

const question = process.argv[2] ?? "What is quantum computing?";

console.log(`Asking: ${question}\n`);

const response = await client.chat("research-agent", question);

console.log("Response:");
console.log(response);

File: sdks/python/examples/basic_chat.py
"""Basic chat example — send a message and print the response."""

import asyncio

from superagent import SuperagentClient


async def main() -> None:
    async with SuperagentClient(base_url="http://localhost:8888") as client:
        response = await client.chat("research-agent", "What is quantum computing?")
        print(response)


if __name__ == "__main__":
    asyncio.run(main())
